tsconfig.json

```diff
@@ -5,7 +5,7 @@
     "esModuleInterop": true,
     "module": "commonjs",
     "moduleResolution": "node",
-    "target": "es2016",
+    "target": "es2020",
```

Comment:
I am personally okay with this, but it changes the compatibility surface of the SDK, esp. for Node where we'd now be targetting Node 14. As that's the current LTS version, it seems safe but it's still a change. Please update the README.

Reply:
I mean I can just write the types definition for `allSettled` as I already did in matrix-react-sdk if that simplifies things. Though if we're happy with es2020 then we should probably change on both layers, babel will probably be the configuration affecting the final output form anyhow